### PR TITLE
Purchase return 

### PR DESCRIPTION
**Pull request by emran-wedevs**

### CODE REVIEW

File: modules/accounting/api/class-rest-api-purchase-return.php [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+        $this->add_log( $invoice_data, 'add' );
+
+        $response = rest_ensure_response( $invoice_data );
+        $response->set_status( 201 );
+
+        return $response;
+    }
+
+    public function prepare_item_for_database( $request ) {
```
Comment (tanmayjay) [inline modules/accounting/api/class-rest-api-purchase-return.php:226]: Kindly use doc block

File: modules/accounting/assets/src/admin/components/purchase-return/InvoiceCreate.vue [hunk 2]
```diff
@@ -0,0 +1,383 @@
+<template>
+    <div class="wperp-container invoice-create">
+
+
```
Comment (tanmayjay) [inline modules/accounting/assets/src/admin/components/purchase-return/InvoiceCreate.vue:4]: unnecessary empty lines

File: modules/accounting/assets/src/admin/components/purchase-return/InvoiceCreate.vue [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+                </div>
+
+
+                <div class="wperp-modal-body">
+
+                    <form action="" method="post" @submit.prevent="searchVoucher">
+                        <!-- End .header-section -->
+                        <div class="wperp-row">
+                            <div class="wperp-form-group wperp-col-sm-8">
+                                <label>Voucher Number</label>
```
Comment (tanmayjay) [inline modules/accounting/assets/src/admin/components/purchase-return/InvoiceCreate.vue:26]: Kindly use localized/i18n string for labels, placeholders

File: modules/accounting/assets/src/admin/components/purchase-return/InvoiceCreate.vue [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+                                <input type="text" v-model="voucher_no" class="wperp-form-field" placeholder="Sales Voucher Number">
+                            </div>
+                            <div class="wperp-form-group wperp-col-sm-4">
+                                <button type="submit" class="wperp-btn btn--primary voucher-search">Search</button>
```
Comment (tanmayjay) [inline modules/accounting/assets/src/admin/components/purchase-return/InvoiceCreate.vue:30]: Kindly use localized/i18n string for labels, placeholders

File: modules/accounting/assets/src/admin/components/purchase-return/InvoiceCreate.vue [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+        quantityUpdate(item, index){
+
+           if( parseFloat(item.qty) > item.returnable_qty){
+                item.qty =  item.returnable_qty ;
+           }
+        },
+        submitReturn(){
+
+            if(!this.summery.total || this.summery.total < 1){
+                Swal.fire('Please , Select minimum one item', '', 'info')  ;
```
Comment (tanmayjay) [inline modules/accounting/assets/src/admin/components/purchase-return/InvoiceCreate.vue:229]: Kindly use localized/i18n string for labels, placeholders

File: modules/accounting/assets/src/admin/components/purchase-return/PurchasesList.vue [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+                @action:click="onActionClick">
+                <template slot="trn_no" slot-scope="data">
+                    <strong>
+                        <router-link :to="{ name: 'purchasesReturnDetails', params: { id: data.row.id }}">
+                            #{{ data.row.id }}
+                        </router-link>
+                    </strong>
+                </template>
+                <template slot="type" slot-scope="data">
+                   Purchase Return
```
Comment (tanmayjay) [inline modules/accounting/assets/src/admin/components/purchase-return/PurchasesList.vue:39]: Kindly use localized/i18n string for labels, placeholders

File: modules/accounting/includes/functions/purchase-return.php [hunk 7]
```diff
@@ -0,0 +1,531 @@
...
+    WHERE purchase.voucher_no = %d",
+        $invoice_no
+    );
+
+    $row = $wpdb->get_row( $sql, ARRAY_A );
+
+    $row['line_items']  = erp_acct_format_invoice_line_items_for_return( $invoice_no );
+    $row['tax_rate_id'] = erp_acct_get_default_tax_rate_name_id();
+
+    $row['attachments'] = unserialize( $row['attachments'] );
```
Comment (tanmayjay) [inline modules/accounting/includes/functions/purchase-return.php:143]: You may consider using `maybe_unserialize` which initially checks if the data is serialized.

Comment (emran-wedevs) [reply modules/accounting/includes/functions/purchase-return.php:143]: ok

**Pull request closed without merging**